spring-cloud-gateway-server-mvc/src/main/java/org/springframework/cloud/gateway/server/mvc/handler/RestClientProxyExchange.java

```diff
@@ -69,4 +74,46 @@ private static ServerResponse doExchange(Request request, ClientHttpResponse cli
 		return serverResponse;
 	}
 
+	private static int copyResponseBody(ClientHttpResponse clientResponse, InputStream inputStream,
```

Comment:
If you move this code into a Utils class, you can also call it from `ClientHttpRequestFactoryProxyExchange`.

Reply:
I've put it to a util
do you know in which case the RestClient or the ClientHttpRequestFactory is being used?

for all my routes it is always the RestClient, maybe if you inject a client manually?

Reply:
`GatewayServerMvcAutoConfiguration` creates the `RestClientProxyExchange`, but with the annotation `@ConditionalOnMissingBean(ProxyExchange.class)`. So if you create any bean that implements `ProxyExchange`, that will be used instead.